### PR TITLE
things with no names get assigned random name and url

### PR DESCRIPTION
**Pull request by elliottw**
fixes issue https://github.com/conusio/conus/issues/67

### CODE REVIEW

File: src/clj/conus/routes/home.clj [hunk 1]
```diff
@@ -64,9 +64,14 @@
     (st/validate params message-schema)))
 
 (defn fix-params [params random-prefix]
+  (let [_ (log/info "poo:" (clojure.string/trim (:name params)))])
```
Comment (zlrth) [inline src/clj/conus/routes/home.clj]: exploratory logging can be removed

File: src/clj/conus/routes/home.clj [hunk 2]
```diff
@@ -64,9 +64,14 @@
...
   (as-> params $
-    (assoc $ :imageurl (if (not (empty? (get-in params [:file :filename]))) (str "/images/" random-prefix (get-in params [:file :filename]))))
-    (assoc $ :name (clojure.string/trim (:name $)))))
+    (assoc $ :imageurl (if (not (empty? (get-in params [:file :filename])))
+                         (str "/images/" random-prefix (get-in params [:file :filename]))))
+    (assoc $ :name (if (not (empty? (clojure.string/trim (:name $))))
+                     (clojure.string/trim (:name $))
+                     (str random-prefix "untitled")
```
Comment (zlrth) [inline src/clj/conus/routes/home.clj]: you arrived at a solution that is semantically simple. 🌞

**Comment by elliottw** [comment]
i wanted to do something more elegant with not-empty, but it didn't work out. also sorry for not remembering the clean up again.

**Comment by zlrth** [comment]
np.
yeah. the simple solution is fine for now. an elegant solution would be integrated with validation and uniqueness-checks. there should be a set of `name` functions. right now there's logic everywhere. bleh!
your solution fits the bill because it gives uniqueness, and is such an ugly name (lolz) that users will be encouraged to change it.

**Pull request merged**